Easydict/Swift/Service/Gemini/GeminiService.swift

```diff
@@ -123,11 +122,116 @@ public final class GeminiService: LLMStreamService {
         Defaults[.geminiAPIKey] ?? ""
     }
 
+    override var availableModels: [String] {
+        Defaults[.geminiValidModels]
+    }
+
+    override var model: String {
+        get {
+            Defaults[.geminiModel]
+        }
+        set {
+            // easydict://writeKeyValue?EZGeminiModelKey=gemini-1.5-flash
+            Defaults[.geminiModel] = newValue
+        }
+    }
+
     // MARK: Private
 
     // Set Gemini safety level to BLOCK_NONE
     private let harassmentBlockNone = SafetySetting(harmCategory: .harassment, threshold: .blockNone)
     private let hateSpeechBlockNone = SafetySetting(harmCategory: .hateSpeech, threshold: .blockNone)
     private let sexuallyExplicitBlockNone = SafetySetting(harmCategory: .sexuallyExplicit, threshold: .blockNone)
     private let dangerousContentBlockNone = SafetySetting(harmCategory: .dangerousContent, threshold: .blockNone)
+
+    private func promptContent(
+        queryType: EZQueryTextType,
+        text: String,
+        from sourceLanguage: Language,
+        to targetLanguage: Language
+    )
+        -> [ModelContent] {
+        var prompts = [[String: String]]()
+
+        switch queryType {
+        case .dictionary:
+            prompts = dictMessages(word: text, sourceLanguage: sourceLanguage, targetLanguage: targetLanguage)
+        case .sentence:
+            prompts = sentenceMessages(sentence: text, from: sourceLanguage, to: targetLanguage)
+        case .translation:
+            fallthrough
+        default:
+            prompts = translationMessages(text: text, from: sourceLanguage, to: targetLanguage)
+        }
+
+        var chats: [ModelContent] = []
+        for prompt in prompts {
+            if let roleRaw = prompt["role"],
+               let parts = prompt["content"] {
+                let role = getCorrectParts(from: roleRaw)
+                let chat = ModelContent(role: role, parts: parts)
+                chats.append(chat)
+            }
+        }
+        guard !chats.isEmpty else {
+            return chats
+        }
+        // removing first element in [ModelContent] since it's system instruction
+        chats.removeFirst()
+        return chats
+    }
+
+    /// Given a roleRaw, replace "assistant" with "model"
+    private func getCorrectParts(from roleRaw: String) -> String {
+        if roleRaw.lowercased() == "assistant" {
+            "model"
+        } else {
+            roleRaw
+        }
+    }
+
+    private func handleResult(
```

Comment:
This part of the code is very similar to the `handleResult` function in `BaseOpenAIService`. 

We can try to optimize it.

Reply:
I moved `handleResult` to `extension LLMStreamService` since it can be used in both `GeminiService` and `OpenAIStreamService`

Reply:
ok